ex_cubic_ingestion/lib/ex_cubic_ingestion/validators.ex

```diff
@@ -0,0 +1,26 @@
+defmodule ExCubicIngestion.Validators do
+  @moduledoc """
+  Module for holding helpful functions for validation
+  """
+
+  @spec is_valid_iso_date?(String.t()) :: boolean()
+  def is_valid_iso_date?(date_str) do
+    match?({:ok, _date}, Date.from_iso8601(date_str))
+  end
+
+  @spec is_valid_iso_datetime?(String.t()) :: boolean()
+  def is_valid_iso_datetime?(datetime_str) do
+    match?({:ok, _datetime}, Timex.parse(datetime_str, "{ISO:Extended}"))
+  end
+
+  @spec is_valid_dmap_dataset_url?(String.t()) :: boolean()
+  def is_valid_dmap_dataset_url?(url) do
+    URI.parse(url).scheme == "https" &&
+      URI.parse(url).path not in [nil, "/"]
```

Comment:
```suggestion
    case URI.parse(url) do
        %{scheme: "https",
             path: valid_path} when is_binary(valid_path) and valid_path != "/" -> true
          %{} -> false
      end
```
alternatively, put the parsed `URI` into a variable instead of parsing it twice.

Reply:
Opted for the variable.